docs/development/architecture.md

```diff
@@ -0,0 +1,221 @@
+- **plugins/examples**: Examples of how to achieve.
+- **docs/**: Documentation that is in markdown, and gets put on the Headlamp Website.
+
+The following components are in separate github repos:
+- **Plugins**: Extensible modules that add custom functionality to the UI. The Headlamp team maintains their plugins in the [headlamp-k8s/plugins repo](https://github.com/headlamp-k8s/plugins). These include plugins for projects like Flux, Backstage and Inspektor Gadget.
+- **Headlamp Website**: Maintained in the [headlamp-k8s/website repo](https://github.com/headlamp-k8s/website). This contains things like the blog and the documentation. The website can be found at https://headlamp.dev/
+
+### Makefile task entry point
+
+The headlamp/ repo [Makefile](https://github.com/kubernetes-sigs/headlamp/blob/main/Makefile) contains targets for building and testing different components.
+
+Here are some examples:
+
+```shell
+make backend
+make backend-lint
+make backend-test
+make run-backend
+make frontend
+make frontend-lint
+make frontend-test
+make run-frontend
+```
+
+### Frontend
+
+- Entry point is [frontend/src/index.tsx](https://github.com/kubernetes-sigs/headlamp/blob/main/frontend/src/index.tsx)
+- See the [Frontend Documentation](https://headlamp.dev/docs/latest/development/frontend) for more information including a Quickstart.
+
+A few key dependencies:
+
+- **TypeScript**: Written in TypeScript.
+- **React Components**: Modular and reusable components that form the UI.
+- **Mui**: For UI components it uses Mui a Material UI component library.
+- **State Management**: Uses Redux for managing application state.
+- **Routing**: Utilizes React Router for navigation within the app.
+- **Npm**: Uses npm for package management.
+
+For each component we have a .stories.tsx file. For example Tables.stories.tsx for the Table.tsx component.
+
+### Backend (headlamp-server)
+
+- The [backend/](https://github.com/kubernetes-sigs/headlamp/blob/main/backend/) folder contains the source code for the headlamp-server binary.
+- Entry point is [backend/cmd/headlamp.go](https://github.com/kubernetes-sigs/headlamp/blob/main/backend/cmd/headlamp.go)
+- See the [Backend Documentation](https://headlamp.dev/docs/latest/development/backend/) for more information including a Quickstart.
+- **Golang**: Written in the [Go programming language](https://golang.org/).
+- **Gorilla**: Uses the Gorilla web toolkit for HTTP routing and middleware.
+- **Kubernetes and Helm**: Integrates with Kubernetes and Helm Go packages.
+
+Packages are stored in the [backend/pkg](https://github.com/kubernetes-sigs/headlamp/blob/main/backend/pkg/) folder. The packages available are cache, config, helm, kubeconfig, logger, plugins, portforward, and utils. Future code should be organized in packages. Some code in the cmd/ folder should move into separate packages.
+
+For each module we have a test file. For example config.go comes with config_test.go.
+
+### App
```

Comment:
I feel like App is a bit too generic term, maybe we can specify that it is a Desktop App?

Reply:
Thanks, fixed. It's Desktop App now.